Fix the access denied by adding a useRef to the queryFn

ChengYanJin opened the pull request:
This will fix the issue of access denied but doesn't require adding token to queryKey
![image](https://github.com/scality/zenko-ui/assets/18453133/07c687cc-2098-43eb-b61d-d017ef7fe43f)

![image](https://github.com/scality/zenko-ui/assets/18453133/34c0cd12-2807-4d61-a1f4-58d8c3e00310)

ChengYanJin committed:
Add useRef to memorize the queryFn to make sure to get the correct one when the token changed

bert-e commented:
# Hello chengyanjin,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/zenko-ui/doc/user).





<details>
  <summary><b>Available options</b></summary>

  name   | description  | privileged   | authored
  ------ | ------------ | ------------ |----------
  `/after_pull_request` | Wait for the given pull request id to be merged before continuing with the current one. |  | 
  `/bypass_author_approval` | Bypass the pull request author's approval |  :star:  | 
  `/bypass_build_status` | Bypass the build and test status |  :star:  | 
  `/bypass_commit_size` | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
  `/bypass_incompatible_branch` | Bypass the check on the source branch prefix |  :star:  | 
  `/bypass_jira_check` | Bypass the Jira issue check |  :star:  | 
  `/bypass_peer_approval` | Bypass the pull request peers' approval |  :star:  | 
  `/bypass_leader_approval` | Bypass the pull request leaders' approval |  :star:  | 
  `/approve` | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
  `/create_pull_requests` | Allow the creation of integration pull requests. |  | 
  `/create_integration_branches` | Allow the creation of integration branches. |  | 
  `/no_octopus` | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
  `/unanimity` | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
  `/wait` | Instruct Bert-E not to run until further notice. |  | 
  

</details>



<details>
  <summary><b>Available commands</b></summary>

  name   | description  | privileged
  ------ | ------------ | ------------
  `/help` | Print Bert-E's manual in the pull request. | 
  `/status` | Print Bert-E's current status in the pull request ```TBA``` | 
  `/clear` | Remove all comments from Bert-E from the history ```TBA``` | 
  `/retry` | Re-start a fresh build ```TBA``` | 
  `/build` | Re-start a fresh build ```TBA``` | 
  `/force_reset` | Delete integration branches & pull requests, and restart merge process from the beginning. | 
  `/reset` | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
  
</details>




*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @hervedombya

* @monpote

* @chengyanjin

* @jbwatenbergscality

ChengYanJin committed:
Throw the error of demical without needing convert the number to string

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @hervedombya

* @monpote

* @jbwatenbergscality

* @chengyanjin

ChengYanJin committed:
Fix the style of edit max capacity modal

ChengYanJin committed:
Fix the recommended max capacity to 80% protected data

ChengYanJin committed:
Fix create veeam account issue

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbwatenbergscality

* @hervedombya

* @chengyanjin

* @monpote

ChengYanJin committed:
Hide the veeam welcome modal for OVA

Since we have two different modals (trial license and Veeam assistant) to avoid them stack together

ChengYanJin committed:
Fix the typing error of window.shellUIRemoteEntryUrl

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbwatenbergscality

* @chengyanjin

* @hervedombya

* @monpote

ChengYanJin committed:
Improve the exit modal in case of failure

JBWatenbergScality commented on src/react/next-architecture/ui/AlertProvider.tsx:
```diff
@@ -0,0 +1,128 @@
+  parents?: string[];
+  selectors?: string[];
+} & { [labelName: string]: string };
+const alertGlobal: any = {};
```
oups any 😢

JBWatenbergScality commented on src/react/ui-elements/Veeam/VeeamWelcomeModal.tsx:
```diff
@@ -16,34 +16,48 @@ import { setSessionState } from '../../utils/localStorage';
   );
   const { isNextLogin } = useNextLogin();
+  const isTrialLicenseModalDisplayed =
+    alerts.length > 0 && props.isFirstTimeLogin;
```
I think we need to add `&& isPlatformAdmin`

JBWatenbergScality commented on src/react/ui-elements/Veeam/useCapacityUnit.ts:
```diff
@@ -21,5 +29,5 @@ export const getCapacityBytes = (
   capacityUnit: string,
 ) => {
-  return (parseInt(capacityValue, 10) * parseInt(capacityUnit, 10)).toString();
+  return (parseFloat(capacityValue) * parseInt(capacityUnit, 10)).toString();
```
Can revert to parseInt

ChengYanJin committed:
Address the comments

ChengYanJin commented:
/approve

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbwatenbergscality

* @chengyanjin

* @hervedombya

* @monpote










*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.2`



The following branches will **NOT** be impacted:

* `development/1.4`
* `development/1.5`
* `development/1.6`
* `development/2.0`
* `development/2.1`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.2`



The following branches have **NOT** changed:

* `development/1.4`
* `development/1.5`
* `development/1.6`
* `development/2.0`
* `development/2.1`



Please check the status of the associated issue ARTESCA-11796.

Goodbye chengyanjin.
